### PR TITLE
coverage: be less smart

### PR DESCRIPTION
**Pull request by mseri**
Copy over all the bisect files, include all folders and avoid using arrays.

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

File: .coverage.sh [hunk 1]
```diff
@@ -8,23 +8,33 @@ mkdir -p $COVERAGE_DIR
...
 jbuilder runtest
 
-outs=$(find . | grep bisect.*.out)
-bisect-ppx-report -I $(dirname $outs[1]) -text report $outs
-bisect-ppx-report -I $(dirname $outs[1]) -summary-only -text summary $outs
-if [ -n "$HTML" ]; then bisect-ppx-report -I $(dirname $outs[1]) -html ../html-report $outs; fi
+# move all bisect files here
+find . -name 'bisect*.out' -exec mv {} . \;
+
+DIRECTORIES=`find . -type d`
```
Comment (lindig) [inline .coverage.sh]: This is quite a nice pattern. The `printf` creates a string "-I xx -I yy -I zz" - one entry per directory.

File: .coverage.sh [hunk 2]
```diff
@@ -22,7 +22,7 @@ jbuilder runtest
 # move all bisect files here
 find . -name 'bisect*.out' -exec mv {} . \;
 
-DIRECTORIES=`find . -type d`
+DIRECTORIES=$(find _build/default -type d -not -path '*/\.*')
```
Comment (lindig) [inline .coverage.sh]: In `'*/\.*'` the `\` is not required but does no harm.

File: .coverage.sh [hunk 3]
```diff
@@ -8,23 +8,33 @@ mkdir -p $COVERAGE_DIR
...
 jbuilder runtest
 
-outs=$(find . | grep bisect.*.out)
-bisect-ppx-report -I $(dirname $outs[1]) -text report $outs
-bisect-ppx-report -I $(dirname $outs[1]) -summary-only -text summary $outs
-if [ -n "$HTML" ]; then bisect-ppx-report -I $(dirname $outs[1]) -html ../html-report $outs; fi
+# move all bisect files here
+find . -name 'bisect*.out' -exec mv {} . \;
+
+DIRECTORIES=$(find _build/default -type d -not -path '*/\.*')
```
Comment (edwintorok) [inline .coverage.sh]: Not sure what this is meant to exclude, are there some hidden directories under _build?

Comment (mseri) [reply .coverage.sh]: Yes, .ppx with the rewriters. It is not necessary though, just nicer output

File: .coverage.sh (hunk 2 above)
Comment (mseri) [reply .coverage.sh]: I wasn’t sure. Do you want me to remove it?

Comment (lindig) [reply .coverage.sh]: Unless you update this PR for something else, this is fine.

**Comment by mseri** [comment]
Aargh I didn't do the squash :( how should we deal with it? Squash and force push locally or keep it like that?

**Comment by lindig** [comment]
I see no problem keeping this.

**Pull request merged**